waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -477,10 +470,25 @@ method init*(g: OnchainGroupManager): Future[void] {.async.} =
                        "could not get the membership deposit: " & getCurrentExceptionMsg())
   g.membershipFee = some(membershipFee)
 
+  var deployedBlockNumber: Uint256
+  try:
+    deployedBlockNumber = await rlnContract.deployedBlockNumber().call()
+  except CatchableError:
+    raise newException(ValueError, 
+                       "could not get the deployed block number: " & getCurrentExceptionMsg())
+  g.rlnContractDeployedBlockNumber = cast[BlockNumber](deployedBlockNumber)
+
   ethRpc.ondisconnect = proc() =
     error "Ethereum client disconnected"
-    let fromBlock = g.latestProcessedBlock.get()
+    let fromBlock = max(g.latestProcessedBlock, g.rlnContractDeployedBlockNumber)
     info "reconnecting with the Ethereum client, and restarting group sync", fromBlock = fromBlock
+    try:
+      let newEthRpc = waitFor newWeb3(g.ethClientUrl)
```

Comment:
I wonder if makes sense to perform some retrial mechanism?

Reply:
hi! https://github.com/waku-org/nwaku/issues/1958 tracks handling delays, this piece of code does handle the reconnections!